module/VuFind/src/VuFind/ILS/Driver/PAIA.php

```diff
@@ -1774,6 +1787,43 @@ protected function paiaPostAsArray($file, $data)
      */
     protected function paiaLogin($username, $password)
     {
+        // as PAIA supports two authentication methods (defined as grant_type:
+        // password or client_credentials), check which one is configured
+        if (!in_array($this->grantType, ['password', 'client_credentials'])) {
+            throw new ILSException(
+                'Unsupported PAIA grantType configured.'
```

Comment:
Might be worth including the illegal grant type value in the exception for easy reference.

Reply:
Good point, I added the illegal grant_type.